Lib/site.py

```diff
@@ -525,11 +525,7 @@ def register_readline():
             pass
 
         def write_history():
-            try:
-                # _pyrepl.__main__ is executed as the __main__ module
-                from __main__ import CAN_USE_PYREPL
-            except ImportError:
-                CAN_USE_PYREPL = False
+            from _pyrepl.main import CAN_USE_PYREPL
```

Comment:
I am not sure this will work correctly under all circumstances. If this module was not imported before we get here, than CAN_USE_PYREPL might have wrong value.

I think really here we should use something like `IS_USING_PYREPL` which will default to false and will only be set to true once the pyrepl is actually initialized and used.

Reply:
Thanks for your feedback 🙂 

> I am not sure this will work correctly under all circumstances. If this module was not imported before we get here, than CAN_USE_PYREPL might have wrong value.

I think I understand what you're saying conceptually, but correct me if I'm wrong. `CAN_USE_PYREPL` defaults to True if you're not on win32. However, you could not be on win32 but still not be able to use the pyrepl and if `interactive_console` wasn't run and didn't get a chance to update it to False in that case, then when we get to this line in `site.py`, we'd have an incorrect value of `CAN_USE_PYREPL`.

Is there a good test case we could write around your concern?

> I think really here we should use something like IS_USING_PYREPL which will default to false and will only be set to true once the pyrepl is actually initialized and used.

This makes sense to me, updated the PR 👍

Reply:
> Is there a good test case we could write around your concern?

You're right that my concern was largely conceptual, I am not sure if (currently) can happen in practice. I suspect it might happen when you set the `PYTHON_BASIC_REPL` variable, although in this particular code snippet it wouldn't make a difference since the if condition below checks for this case specifically.